cedar-drt/fuzz/src/schemas.rs

```diff
@@ -64,8 +64,29 @@ fn action_type_equivalence(name: &str, lhs: ActionType, rhs: ActionType) -> Resu
     } else {
         match (lhs.applies_to, rhs.applies_to) {
             (None, None) => Ok(()),
+            (Some(lhs), Some(rhs)) if is_either_empty(&lhs) => {
+                if is_either_empty(&rhs) {
+                    Ok(())
+                } else {
+                    Err(format!(
+                        "Mismatched applies to in `{name}`. lhs : `{:?}`,rhs: `{:?}`",
+                        lhs, rhs
+                    ))
+                }
+            }
+            (Some(lhs), Some(rhs)) if is_either_empty(&rhs) => {
+                if is_either_empty(&lhs) {
+                    Ok(())
+                } else {
+                    Err(format!(
+                        "Mismatched applies to in `{name}`. lhs : `{:?}`,rhs: `{:?}`",
+                        lhs, rhs
+                    ))
+                }
+            }
+            // if neither is non-applicable, they must be equal
             (Some(lhs), Some(rhs)) => {
-                if empty_target(&rhs) || empty_target(&lhs) || rhs == lhs {
+                if rhs == lhs {
```

Comment:
Splitting up these cases made the code difficult to read. I think it would be cleaner to have something like:
```rust
(Some(lhs), Some(rhs)) => {
    if is_either_empty(&lhs) && is_either_empty(&rhs) {
        Ok(())
    } else if rhs == lhs {
        Ok(())
    } else {
    ...
```

Reply:
Note the proposed refactor changes behavior. I'd recommend:
```
if lhs == rhs {
  Ok(())
} else if empty_target(&rhs) || empty_target(&lhs) {
   ...
}
```

Reply:
I think the refactor is the correct behavior here: if one of the LHS or RHS has an empty `appliesTo` list, then both should.

Reply:
I ended up adopting Kesha's refactor because it's easier to understand (please refer to the comments).